app/src/androidTest/java/nextstep/shoppingcart/CartListScreenTest.kt

```diff
@@ -0,0 +1,179 @@
+package nextstep.shoppingcart
+
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.toMutableStateList
+import androidx.compose.ui.test.isDisplayed
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithContentDescription
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import nextstep.shoppingcart.ext.getFormattedPrice
+import nextstep.shoppingcart.model.Cart
+import nextstep.shoppingcart.model.Product
+import nextstep.shoppingcart.ui.cart.CartListScreen
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+
+class CartListScreenTest {
+
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    @Test
+    fun `주어진_CartList의_전체_가격이_올바르게_나와야_한다`() {
+        //given
+        val cartList = listOf(
+            Cart(Product(1, "상품1", 1000, ""), 1),
+            Cart(Product(2, "상품2", 2000, ""), 2),
+            Cart(Product(3, "상품3", 3000, ""), 3),
+            Cart(Product(4, "상품4", 4000, ""), 4),
+        )
+        //when
+        composeTestRule.setContent {
+            CartListScreen(cartList)
+        }
+
+        //then
+        composeTestRule
+            .onNodeWithText("주문하기(${cartList.sumOf { it.totalPrice }.getFormattedPrice()})")
+            .assertExists()
+            .isDisplayed()
+    }
+
+    @Test
+    fun `주어진_CartList에서_특정상품의_갯수를_추가하였을때_올바르게_추가되어야_한다`() {
+        //given
+        var cartListState = mutableStateListOf<Cart>().apply {
+            addAll(
+                listOf(
+                    Cart(Product(1, "상품1", 1000, ""), 1),
+                    Cart(Product(2, "상품2", 2000, ""), 2),
+                    Cart(Product(3, "상품3", 3000, ""), 3),
+                    Cart(Product(4, "상품4", 4000, ""), 4),
+                )
+            )
+        }
```

Comment:
```suggestion
        var cartListState = mutableStateListOf<Cart>(
            Cart(Product(1, "상품1", 1000, ""), 1),
            Cart(Product(2, "상품2", 2000, ""), 2),
            Cart(Product(3, "상품3", 3000, ""), 3),
            Cart(Product(4, "상품4", 4000, ""), 4),
        )
        
        // 위 내용과 별개로 State일 필요가 없는 List인 경우 이런 형태도 가능합니다.
        var cartListState = MutableList(4) { index ->
            Cart(Product(index, "상품$index", 1000 * index, ""), index)
        }
```

apply와 직접 allAll을 안해도 이런 방식으로 더미 데이터를 만들 수 있습니다.

Reply:
더 가독성있고 좋아보이네요. 

개인적으로 테스트에 사용하는데 더 적합한건 위 방식 같습니다!

Reply:
a3d8650217630ff22fa42eba2e1d055d52968201